apps/hash-frontend/src/pages/account/integrations/linear/new.page.tsx

```diff
@@ -1,100 +1,153 @@
+
+  const linearIntegration = useMemo(() => {
+    if (linearIntegrations && linearIntegrationEntityId) {
+      return linearIntegrations.find(
+        ({ entity }) =>
+          entity.metadata.recordId.entityId === linearIntegrationEntityId,
+      );
+    }
+  }, [linearIntegrations, linearIntegrationEntityId]);
 
   useEffect(() => {
     void (async () => {
-      const { data } = await queryEntities({
-        data: {
-          operation: {
-            multiFilter: {
-              filters: [
-                /** @todo: figure out how to make this work */
-                // {
-                //   field: ["ownedById"],
-                //   operator: "EQUALS",
-                //   value: authenticatedUser.accountId,
-                // },
-                {
-                  field: ["metadata", "entityTypeId"],
-                  operator: "EQUALS",
-                  value: types.entityType.linearIntegration.entityTypeId,
-                },
-              ],
-              operator: "AND",
-            },
-          },
-        },
-      });
+      if (linearIntegration && linearIntegrations) {
+        const linearOrgId = linearIntegration.entity.properties[
+          extractBaseUrl(types.propertyType.linearOrgId.propertyTypeId)
+        ] as string;
+
+        const { data } = await getLinearOrganization({
+          variables: { linearOrgId },
+        });
 
-      if (data) {
-        setLinearIntegrationEntities(getRoots(data));
+        if (data) {
+          const organization = data.getLinearOrganization;
+
+          setLinearOrganization(
+            mapLinearOrganizationToLinearOrganizationTeamsWithWorkspaces({
+              linearIntegrations,
+            })(organization),
+          );
+        } else {
+          throw new Error("Could not get linear organization");
+        }
       }
     })();
-  }, [queryEntities, authenticatedUser]);
-
-  const linearIntegrationEntity = useMemo(() => {
-    if (linearIntegrationEntities && linearIntegrationEntityId) {
-      return linearIntegrationEntities.find(
-        ({
-          metadata: {
-            recordId: { entityId },
-          },
-        }) => entityId === linearIntegrationEntityId,
-      );
-    }
-  }, [linearIntegrationEntities, linearIntegrationEntityId]);
+  }, [linearIntegrations, linearIntegration, getLinearOrganization]);
+
+  const possibleWorkspaces = useMemo(
+    () => [authenticatedUser, ...authenticatedUser.memberOf],
+    [authenticatedUser],
+  );
 
-  const linearOrgId = linearIntegrationEntity?.properties[
-    extractBaseUrl(types.propertyType.linearOrgId.propertyTypeId)
-  ] as string | undefined;
+  const handleSaveAndContinue = useCallback(async () => {
+    if (linearIntegration && linearOrganization) {
+      /** @todo: add proper error handling */
+
+      await syncLinearIntegrationWithWorkspaces({
+        variables: {
+          linearIntegrationEntityId:
+            linearIntegration.entity.metadata.recordId.entityId,
```

Comment:
I think these must be equivalent since you find one from the other
```suggestion
          linearIntegrationEntityId,
```

Reply:
That's true, but `linearIntegrationEntityId` is populated form a query param so I felt it's safer to use the one from the `linearIntegration` as it's fetched from the server.

Reply:
`linearIntegration` is the result of this:
```ts
linearIntegrations.find(
  ({ entity }) =>
    entity.metadata.recordId.entityId === linearIntegrationEntityId,
);
```
unless I'm missing something, it is logically impossible for them to be different

Reply:
If the `linearIntegrationEntityId` query param is something incorrect, then `linearIntegration` will remain undefined and the `useEffect` won't try to fetch the organizations with an incorrect ID. If I make use of `linearIntegrationEntityId` directly in the `useEffect` will try to fetch the organizations with the incorrect ID. 

But I suppose neither scenario is great anyway, we'll want to handle that scenario more gracefully perhaps by redirecting back to the `/account/integrations` page for example.

If I modify the `useMemo` hook to do that would you be happy with it?
```TypeScript
  const linearIntegration = useMemo(() => {
    if (linearIntegrations && linearIntegrationEntityId) {
      const integration = linearIntegrations.find(
        ({ entity }) =>
          entity.metadata.recordId.entityId === linearIntegrationEntityId,
      );
      
      if (!integration) {
        void router.push("/account/integrations");
      }

      return integration;
    }
  }, [linearIntegrations, linearIntegrationEntityId, router]);
```

Reply:
I'm suggesting that you could do the find for `linearIntegration` inside the same effect where you fetch the org, since it's only required there.
```ts
   useEffect(() => {
    void (async () => {
      if (!linearIntegrations || !linearIntegrationEntityId) {
        return;
      }

      const linearIntegration = linearIntegrations.find(
        ({ entity }) =>
          entity.metadata.recordId.entityId === linearIntegrationEntityId,
      );

      if (!linearIntegration) {
        void router.push("/account/integrations");
        return;
      }

      const linearOrgId = linearIntegration.entity.properties[
        extractBaseUrl(types.propertyType.linearOrgId.propertyTypeId)
      ] as string;

      const { data } = await getLinearOrganization({
        variables: { linearOrgId },
      });

      if (data) {
        const organization = data.getLinearOrganization;

        setLinearOrganization(
          mapLinearOrganizationToLinearOrganizationTeamsWithWorkspaces({
            linearIntegrations,
          })(organization),
        );
      } else {
        throw new Error("Could not get linear organization");
      }
    })();
  }, [
    linearIntegrations,
    linearIntegrationEntityId,
    getLinearOrganization,
    router,
  ]);
```

Reply:
I see how that's cleaner now, thank for explaining. Have addressed in https://github.com/hashintel/hash/pull/2826/commits/0afd9f20d7568e28b60e92a2fa125ea628fec88a

Reply:
For the save callback, I think you'd need to change the `EntityId` you're importing to satisfy the type
```ts
import { EntityId } from "@local/hash-subgraph";

...

  const linearIntegrationEntityId = useMemo(() => {
    return router.query.linearIntegrationEntityId as EntityId | undefined; // add | undefined since it might be
  }, [router]);

...

  const handleSaveAndContinue = useCallback(async () => {
    if (linearOrganization && linearIntegrationEntityId) {
      /** @todo: add proper error handling */

      await syncLinearIntegrationWithWorkspaces({
        variables: {
          linearIntegrationEntityId,
          syncWithWorkspaces:
            mapLinearOrganizationToSyncWithWorkspacesInputVariable({
              linearOrganization,
              possibleWorkspaces,
            }),
        },
      });

      void router.push("/account/integrations/linear");
    }
  }, [
    syncLinearIntegrationWithWorkspaces,
    linearIntegrationEntityId,
    linearOrganization,
    possibleWorkspaces,
    router,
  ]);
```